privacyidea/lib/importotp.py

```diff
@@ -540,7 +573,12 @@ def parsePSKCdata(xml_data,
                 token["timeShift"] = key.data.timedrift.text.strip()
 
         tokens[serial] = token
-    return tokens
+
+    if abort:
+        not_imported_serials = [serial for serial in tokens]
```

Comment:
If you want to get the list of the keys from a dictionary, you can simply do this like this:

    not_imported_serials = list(tokens)